### PR TITLE
Review Release Notes Run on External PRs

### PR DESCRIPTION
**Pull request by kgal-pan**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-4971

## Description
Same changes introduced in https://github.com/demisto/content/pull/19199 to allow running Release Notes Review on external PRs.

## Tests

Used [external PR](https://github.com/demisto/content/pull/24072) from fork https://github.com/kgal-pan/content to test.

#### Community Pack Bump

Bumped `HelloWorld` community Pack version and invalid release note.

**Expected:**
- Review Release Notes job to run.
- Release note review is skipped.

**Actual**
- Review Release Notes job ran.
- [Release note review is skipped](https://github.com/demisto/content/actions/runs/3997808430/jobs/6859615870#step:7:18).

#### XSOAR-supported Pack Bump

Bumped `SentinelOne` XSOAR-supported Pack version and empty release note.

**Expected:**
- Review Release Notes job to run.
- Release note is reviewed.

**Actual**
- Review Release Notes job ran.
- [Release note is reviewed](https://github.com/demisto/content/actions/runs/3997972453/jobs/6859998639#step:7:20).

### CODE REVIEW

File: .github/workflows/review-release-notes.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ on: pull_request
 jobs:
   release_notes_review:
     runs-on: ubuntu-latest
```
Comment (ShacharKidor) [inline .github/workflows/review-release-notes.yml:6]: @kgal-pan - are we sure that we need to remove the - github.repository == 'demisto/content' condition? 
As I remember you are probably right, but have you tested it?

File: Tests/secrets_white_list.json [hunk 2]
```diff
@@ -1913,6 +1913,7 @@
       "384:mzawcK1gOUqJMpMhE5yOaF0O9TwFExCZkUf6H+PknKV+rXkrsH/CxEdubVAjMPW4:AiMVWcK9AhrDNEf"
     ],
     "emails": [
+      "changed-files@v18.7",
```
Comment (ShacharKidor) [inline Tests/secrets_white_list.json:1916]: What is this change related to?

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
@kgal-pan - Nice :)
Please test that this works as expected  - open a PR from a forked repo of content (set this branch as the base branch of your PR), make sure that the review rn action is triggered, and add here the relevant links.
Also worth to make sure that it is still triggered for content regular PRs as usual (not from a forked repo).

**Comment by ShacharKidor** [comment]
Related old PR with explanations: [19199](https://github.com/demisto/content/pull/19199).

File: Tests/secrets_white_list.json (hunk 2 above)
Comment (kgal-pan) [reply Tests/secrets_white_list.json:1916]: Because the GitHub Action version string is detected as an email and causing [the validation to fail](https://app.circleci.com/pipelines/github/demisto/content/228267/workflows/1a7c90bd-f45e-4479-b9ac-f942617569d3/jobs/638868). The same was done [here](https://github.com/demisto/content/pull/23306).

File: .github/workflows/review-release-notes.yml (hunk 1 above)
Comment (kgal-pan) [reply .github/workflows/review-release-notes.yml:6]: Tested, details in https://github.com/demisto/content/pull/24061#issuecomment-1402200682.

**Review by ShacharKidor — APPROVED** [review]

**Review by DinaMeylakh — APPROVED** [review]
Cool cool cool

**Comment by kgal-pan** [comment]
@ShacharKidor @ShahafBenYakir - In continuation to our discussion yesterday, 

I added a conditional to check if the destination repo is `content`, retriggered test PR with change to XSOAR-supported Pack (SentinelOne) and confirmed [the job](https://github.com/demisto/content/actions/runs/4053800321/jobs/6974849188#step:7:20) ran and the release notes were reviewed (as expected).
To confirm that the Review Release Notes workflow is not run when a PR is opened against the fork's `master` base branch, I opened https://github.com/kgal-pan/content/pull/234. As expected, [the workflow/job was skipped](https://github.com/kgal-pan/content/actions/runs/4053906411/jobs/6975087774):

image


Since the Review Release Notes workflow specifies that it will only run on PRs, after running the tests described above, we can confirm that the workflow/job will only run on PRs opened to `content`.
